docs/Maintainer-Guidelines: add mission. (#5223)
Homebrew/brew: docs/Maintainer-Guidelines.md

This document is current practice. If you wish to change or discuss any of the below: open a PR to suggest a change.

## Mission

> Homebrew seeks to be the missing package manager for macOS. It's primary goal is to be useful. It should be as useful as possible to as many people as possible that we can maintain as a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to use macOS features and blend in with the macOS and Apple ecosystem.

Reviewer: The third sentence reads weird to me. I understand the meaning, but I feel some non-native english speakers might be confused. It’s the “that” connector that’s tripping me up, I think.

Also, might it me wise to remove the “commercially”? On commercial apps support is supposed to always be nice, even if the customer is rude. Here we typically warn/ask users to be more civil and may even ban them if they’re irredeemably difficult.

> Homebrew seeks to be the missing package manager for macOS. It's primary goal is to be useful. It should be as useful as possible to as many people as possible that we can maintain as a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to use macOS features and blend in with the macOS and Apple ecosystem.

Reviewer 2: ```suggestion
Homebrew seeks to be the missing package manager for macOS. Its primary goal is to be useful. It should be as useful as possible to as many people as possible while being maintainable by a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to use macOS features and blend in with the macOS and Apple ecosystem.
```

Native English speaker here, and yes I agree the third sentence needs re-wording. Sorry for dropping the ball on this @MikeMcQuaid and thanks for adding it!

> Homebrew seeks to be the missing package manager for macOS. It's primary goal is to be useful. It should be as useful as possible to as many people as possible that we can maintain as a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to use macOS features and blend in with the macOS and Apple ecosystem.

Reviewer 3: I've incorporated others' suggestions and added a few edits of my own.
```suggestion
Homebrew aims to be the missing package manager for macOS, whose primary goal is to be useful. It aims to be as useful as possible to as many people as possible, while being maintained by a small group of volunteers to a high standard of quality. Where possible and sensible it should seek to use features of macOS to blend in with the macOS ecosystem.
```

Author: Thanks all, have gone with a blend of @zbeekman and @sjackman's versions.

> Also, might it me wise to remove the “commercially”? On commercial apps support is supposed to always be nice, even if the customer is rude.

@vitorgalvao Pedantic perhaps but I'd say that's not a statement about the quality of Homebrew but the quality of Homebrew's support (which, we agree, should not be required to meet a commercial standard).

Reviewer: > Pedantic perhaps

Not at all, I agree. I was commenting in the sense of not wanting a rude user throwing that in our faces later on.

> Homebrew aims to be the missing package manager for macOS. Its primary goal is to be useful. It should be as useful as possible to as many people as possible while remaining maintainable by a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to features of macOS to blend in with the macOS and Apple ecosystem.

Reviewer 3: ```suggestion
Homebrew aims to be the missing package manager for macOS. Its primary goal is to be useful. It should be as useful as possible to as many people as possible while remaining maintainable by a small group of volunteers to a high standard. Where possible and sensible it should seek to use features of macOS to blend in with the macOS ecosystem.
```

Reviewer 2: LGTM

Author: The "commercially" part is important; it's what defines limiting what we "officially" support to stuff that actually works. The Apple ecosystem part is also important as we want to fit in with the wider Apple ecosystem beyond just macOS.

Reviewer 3: > it's what defines limiting what we "officially" support to stuff that actually works

Even after this clarification, I'm not sure what you mean by "commercially" in this context. I'm not trying to be a pedantic pain; I genuinely don't understand. By "commercially" you mean to say that "we officially support stuff that actually works", as opposed to what? To quote Princess Bride, "I do not think it means what you think it means." =p

Reviewer 4: Does "professionally" have a similar meaning to "commercially" in this context? ie "a professionally high standard"

Author: No, these are good questions and I'm glad they are asked. Many open source projects (including Homebrew in the past) have a "well X kinda works but I wouldn't recommend it" attitude to various features/repositories/etc. That does not fly for commercial organisations and should not fly for us. In short, our benchmark should be software that is being sold for money rather than software that is available free of charge. Similarly, we should not see ourselves as a hobby or club but a organisation that behaves professionally and treats our fellow maintainers as we would coworkers.

Rambly, I know, but hopefully that helps a bit?

Reviewer 3: I understand now. Thanks for the clarification, Mike. I like Steven's suggestion. How about
```
… to a professional standard of quality.
```

> Homebrew aims to be the missing package manager for macOS. Its primary goal is to be useful. It should be as useful as possible to as many people as possible while remaining maintainable by a small group of volunteers to a commercially high standard. Where possible and sensible it should seek to features of macOS to blend in with the macOS and Apple ecosystem.

Reviewer 5: ```suggestion
Homebrew aims to be the missing package manager for macOS. Its primary goal is to be useful to as many people as possible, while remaining maintainable to a high standard by a small group of volunteers. Where possible and sensible, it should seek to use features of macOS to blend in with the macOS ecosystem.
```
